packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderActions/ConditionBuilderActions.js

```diff
@@ -0,0 +1,160 @@
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import { Close, WarningAltFilled } from '@carbon/react/icons';
+import { Section, Heading } from '@carbon/react';
+import { ConditionBuilderItem } from '../ConditionBuilderItem/ConditionBuilderItem';
+import {
+  blockClass,
+  translateWithId,
+} from '../ConditionBuilderContext/DataConfigs';
+import { ItemOption } from '../ConditionBuilderItem/ConditionBuilderItemOption/ItemOption';
+import { ConditionBuilderContext } from '../ConditionBuilderContext/ConditionBuilderProvider';
+import ConditionBuilderAdd from '../ConditionBuilderAdd/ConditionBuilderAdd';
+import uuidv4 from '../../../global/js/utils/uuidv4';
+import { ConditionBuilderButton } from '../ConditionBuilderButton/ConditionBuilderButton';
+import { focusThisField } from '../utils/util';
+
+const ConditionBuilderActions = ({ actions, className }) => {
+  const { actionState, setActionState } = useContext(ConditionBuilderContext);
+
+  const addActionHandler = () => {
+    let action = {
+      id: uuidv4(),
+      label: undefined,
+      popoverToOpen: 'actionField',
+    };
+    setActionState([...actionState, action]);
+  };
+
+  const onchangeHandler = (evt, selectedId, actionIndex) => {
+    let currentAction = actions.filter((action) => action.id == selectedId);
+    let updatedActions = [];
+
+    if (checkDuplicateAction(selectedId)) {
+      let duplicateAction = { id: uuidv4(), label: 'Invalid', invalid: true };
+      updatedActions = [
+        ...actionState.slice(0, actionIndex),
+        duplicateAction,
+        ...actionState.slice(actionIndex + 1),
+      ];
+    } else {
+      updatedActions = [
+        ...actionState.slice(0, actionIndex),
+        ...currentAction,
+        ...actionState.slice(actionIndex + 1),
+      ];
+    }
+    setActionState(updatedActions);
+
+    focusThisField(evt);
+  };
+
+  const checkDuplicateAction = (selectedId) => {
```

Comment:
something like this could be a handy utility function. i would move this into `utils`.

Reply:
modified as advised